Add Bazel Test Optimization setup docs (#37551)
DataDog/documentation: content/en/tests/setup/_index.md

<br>

> If you run Go, Java, or Python tests with Bazel, use Datadog's official [Bazel rules for Test Optimization][5].

Reviewer: ```suggestion
If you use Bazel to run Go, Java, or Python tests, use the Datadog [Bazel rules for Test Optimization][5].
```
